library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ogg/OggExtractor.java

```diff
@@ -90,25 +90,32 @@ public int read(ExtractorInput input, PositionHolder seekPosition) throws IOExce
 
   @EnsuresNonNullIf(expression = "streamReader", result = true)
   private boolean sniffInternal(ExtractorInput input) throws IOException {
-    OggPageHeader header = new OggPageHeader();
-    if (!header.populate(input, true) || (header.type & 0x02) != 0x02) {
-      return false;
-    }
+    OggPageHeader header = new OggPageHeader(-1L);
 
-    int length = min(header.bodySize, MAX_VERIFICATION_BYTES);
-    ParsableByteArray scratch = new ParsableByteArray(length);
-    input.peekFully(scratch.getData(), 0, length);
-
-    if (FlacReader.verifyBitstreamType(resetPosition(scratch))) {
-      streamReader = new FlacReader();
-    } else if (VorbisReader.verifyBitstreamType(resetPosition(scratch))) {
-      streamReader = new VorbisReader();
-    } else if (OpusReader.verifyBitstreamType(resetPosition(scratch))) {
-      streamReader = new OpusReader();
-    } else {
-      return false;
+    while (header.populate(input, true) && (header.type & 0x02) == 0x02) {
+      int length = min(header.bodySize, MAX_VERIFICATION_BYTES);
+      ParsableByteArray scratch = new ParsableByteArray(length);
+      input.peekFully(scratch.getData(), 0, length);
+
+      if (FlacReader.verifyBitstreamType(resetPosition(scratch))) {
+        streamReader = new FlacReader(header.streamSerialNumber);
+        return true;
+      } else if (VorbisReader.verifyBitstreamType(resetPosition(scratch))) {
+        streamReader = new VorbisReader(header.streamSerialNumber);
+        return true;
+      } else if (OpusReader.verifyBitstreamType(resetPosition(scratch))) {
+        streamReader = new OpusReader(header.streamSerialNumber);
+        return true;
+      }
+      try {
+        input.skip((int) input.getPeekPosition());
+        header.skipToNextPage(input);
```

Comment:
Based on the documentation of Extractor.sniff(ExtractorInput input), we can only modify the reading position when returning true from this method. If false is returned only peek position is modified.

We change the read position by calling skip on input and may still return false.